lib/ansible/plugins/filter/core.py

```diff
@@ -289,7 +289,12 @@ def mandatory(a):
             name = "'%s' " % to_text(a._undefined_name)
         else:
             name = ''
-        raise AnsibleFilterError("Mandatory variable %s not defined." % name)
+
+        if msg is not None:
+            raise AnsibleFilterError(str(msg))
```

Comment:
Use to_native instead of str.
```suggestion
            raise AnsibleFilterError(to_native(msg))
```

Reply:
I never got the github alert for this change request for some reason. :sob:  Requested change has been made.  Sorry for the delay.